oarepo_oidc_einfra/resources.py

```diff
@@ -67,18 +89,94 @@ def upload_dump(self):
                 "message": "Content-Type must be application/json",
             }, 400
 
-        dump_url = current_app.config["EINFRA_DUMP_DATA_URL"]
-        now = datetime.utcnow().strftime("%Y-%m-%d-%H-%M-%S")
-        dump_path = f"{dump_url}/{now}.json"
-
-        location = PyFSFileStorage(dump_path)  # handles both filesystem and s3
-        with location.open(mode="wb") as f:
-            f.write(request.data)
-
+        dump_path = store_dump(request.data)
         update_from_perun_dump.delay(dump_path)
         return {"status": "ok"}, 201
 
+    @login_required
+    def accept_invitation(self) -> Response:
+        """Accept an invitation to join a community.
+
+        This is an endpoint to which user is directed
+        after clicking the link in the invitation email, accepting the terms and conditions and
+        accepting the invitation.
+
+        We expect the url to contain the request_id of the invitation request that was sent to the user
+        and use it to accept the invitation.
 
-def create_rest_blueprint(app):
+        Note:
+        If user accepts the invitation but this endpoint is not called, the invitation will be forever
+        in the submitted state (until expiration). The user will still be able to access the community
+        because the AAI will return the correct capabilities for the user.
+
+        Currently, the PERUN api does not return the ID of the created invitation, so we cannot store it
+        and check in a background task if the invitation was accepted and then change the state of the request.
+
+        """
+        assert request.view_args is not None
+
+        request_id = decrypt(request.view_args["request_id"])
+
+        # get the invitation request and check if it is submitted.
+        invitation_request = Request.get_record(request_id)
+        assert invitation_request.state == "submitted"
+
+        # if its user is not the current user, we might to delete
+        # the user on the request so that it does not pollute the space
+        request_user_id = invitation_request.payload.get("user_id")
+        if request_user_id != str(g.identity.id):
+            user = User.query.filter_by(User.id == request_user_id).one()
+            if not user.is_active:
+                db.session.delete(user)
+            else:
+                log.error(
+                    "Invitation check failed: The user for which the invitation was sent (%s) "
+                    "is an active user and is not the same as the current user %s, thus the "
+                    "invitation was not accepted. This means that we need to check the users if duplicity"
+                    "exists and if so, we need to merge them somehow.",
+                    request_user_id,
+                    g.identity.id,
+                )
+                raise PermissionDenied(
+                    "The invitation was intended for a different user"
```

Comment:
I would not expose the fact the link is valid for some other user,  return something rather generic, like BadRequest `Invitation link invalid`